### PR TITLE
build-cache with GHA NPM build

### PR DESCRIPTION
**Pull request by pflynn-virtru**
- adds a build docker cache - https://github.com/moby/buildkit#github-actions-cache-experimental
- adds a node cache
- moves web build into GHA instead of inside container
- fixes some critical bugs that was trigger with CI.build=true in GHA
- reduces build time to 3-9 minutes

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -117,17 +128,19 @@ jobs:
...
           push: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' && github.repository == 'opentdf/frontend' }}
           tags: ${{ steps.meta.outputs.tags }}
           labels: ${{ steps.meta.outputs.labels }}
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
 
   chart:
     runs-on: ubuntu-latest
     needs:
-      - app
```
Comment (dmihalcik-virtru) [inline .github/workflows/build.yaml:130]: This is needed to get the app version info

File: Dockerfile [hunk 2]
```diff
@@ -29,8 +29,9 @@ RUN CGO_ENABLED=0 GOOS=linux go build -v -a  -o /server
 # server - nginx alpine
 FROM scratch as server
 WORKDIR /
-COPY --from=builder /build/build/ /www/
 COPY --from=gobuilder /server /server
+# in CI the build out put is build/, locally it is dist/
+COPY build/ /www/
```
Comment (dmihalcik-virtru) [inline Dockerfile:34]: Will this work on aarch64 (ARM platforms)?

**Review by dmihalcik-virtru — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yaml (hunk 1 above)
Comment (pflynn-virtru) [reply .github/workflows/build.yaml:130]: oh, it was just a duplicate removal.  it still blocks for it

Comment (dmihalcik-virtru) [reply .github/workflows/build.yaml:130]: OIC can you keep these alphabetized?

**Review by dmihalcik-virtru — APPROVED** [review]

File: Dockerfile (hunk 2 above)
Comment (pflynn-virtru) [reply Dockerfile:34]: The contents of `build/` are only web artifacts that are served by the http server written in Go.  The Go server is build for arm and amd. see https://github.com/opentdf/frontend/actions/runs/3858700267/jobs/6577513072#step:10:142

File: .github/workflows/build.yaml (hunk 1 above)
Comment (pflynn-virtru) [reply .github/workflows/build.yaml:130]: done

**Pull request merged**